addons/account_edi/models/account_move.py

```diff
@@ -337,11 +337,12 @@ def _get_edi_document(self, edi_format):
     def _get_edi_attachment(self, edi_format):
         return self._get_edi_document(edi_format).sudo().attachment_id
 
-    # this override is to make sure that the main attachment is not the edi xml otherwise the attachment viewer will not work correctly
-    def _message_set_main_attachment_id(self, attachment_ids):
-        if self.message_main_attachment_id and len(attachment_ids) > 1 and self.message_main_attachment_id in self.edi_document_ids.attachment_id:
-            self.message_main_attachment_id = self.env['ir.attachment']
-        super()._message_set_main_attachment_id(attachment_ids)
+    def _message_set_main_attachment_id(self, attachment_ids, force=False):
+        """ This override is to make sure that the main attachment is not the
+        edi xml otherwise the attachment viewer will not work correctly """
```

Comment:
This is not documentation but a comment, is it? :wink:

Reply:
Eternal fight between comments and docstrings, indeed. Will revert the change as it is not out team addons :) .

Reply:
With this change:
```python
In [1]: help(self.env['account.move']._message_set_main_attachment_id)

"""
Help on method _message_set_main_attachment_id in module odoo.addons.account_edi.models.account_move:

_message_set_main_attachment_id(attachment_ids, force=False) method of odoo.api.account.move instance
    This override is to make sure that the main attachment is not the
    edi xml otherwise the attachment viewer will not work correctly
"""
```
![image](https://github.com/odoo/odoo/assets/44770049/18f248e4-546e-45be-b461-ac23286f415b)


Without it:
```python
In [1]: help(self.env['account.move']._message_set_main_attachment_id)

"""
Help on method _message_set_main_attachment_id in module odoo.addons.account_edi.models.account_move:

_message_set_main_attachment_id(attachment_ids, force=False) method of odoo.api.account.move instance
    Update 'main' attachment.
    
    :param list attachment_ids: new main attachment IDS; if several attachments
      are given, we search for pdf or image first;
    :param boolean force: if set, replace an existing attachment; otherwise
      update is skipped;
"""
```
![image](https://github.com/odoo/odoo/assets/44770049/20289e1a-1764-4a0c-8c36-861152159bf6)
In this case my IDE is not configured to go see the dosctring of the method but at least it becomes possible.

If I comment out the function completely (I didn't to have the output out of ipython) I have the original docstring though
![image](https://github.com/odoo/odoo/assets/44770049/062f6acd-510b-4438-b5d9-700b68928a96)



I really don't think that a IDE or any documentation tool should show that comment instead of proper docstring.